### DISCUSSION TITLE
Shouldn't GoogleSignIn.signOut() trigger a firebase.auth().onAuthStateChanged update?

### DISCUSSION THREAD

**Discussion by cbdeveloper** [invertase/react-native-firebase#7549] [Q&A]
Pretty much the question from the title.

When I do `GoogleSignin.signIn()`, I get an `auth().onAuthStateChanged()` update, as expected.

But when I do, `GoogleSignin.signOut()`, the `onAuthStateChanged` listener is not being triggered, unless I do `auth().signOut()`.

Is that the expected behavior?

Thank you

**Comment by davidepalazzo** [comment]
Hi,

You shouldn't need to call `GoogleSignin.signOut()` if you are using `auth().signInWithCredential(googleCredential)`, as shown in the docs: [Social Auth for Google](https://rnfirebase.io/auth/social-auth#google). Once the credential is passed to the `auth()` instance, it takes care of handling the user's session and `auth().signOut()` is what you need to invoke.

In case you aren't already using `auth().signInWithCredential`, I'd suggest giving it a try!

**Reply by cbdeveloper** [reply]
Hey @davidepalazzo , indeed, that makes sense. When I call `auth().signOut`, I do get the `onAuthStateChanged` update.

But another thing has caught my attention. If I just do `auth().signOut()`, when I sign back in, I don't get prompted for selecting the Google account again. It's signs in automatically on the previous logged in account.

Only if I do `GoogleSignin.signOut()` (on top of the `auth().signOut()`), is when I get prompted for selecting the account again, which is what I need. I don't want it to sign me in without explicitly prompting me for selecting the account, even if I had signed in/out before with that account.

Is that expected?

**Reply by davidepalazzo** [reply]
Yes, that's right. `auth()` controls the Firebase Auth flow and doesn't clear the session for the underlying IDP.

A potential drawback of using `GoogleSignin.signOut()` is that it will also sign the user out from their default browser if they were logged in.

**Reply by mikehardy** [reply]
🤔 

1) only prompting for new google account selection if you sign out from underlying google account is expected behavior

2) not getting auth event when you sign in underlying google account may be a bug in firebase-ios-sdk but I think rather not - until you pass the token/credential from the google sign in into firebase auth, firebase does not know about it so it is not an auth state change. However when you then sign in for firebase auth or link the credential from the google sign in, it should trigger an event

It sounds like things are working as expected ?

**Reply by cbdeveloper** [reply]
Yeah, it looks like it's working as expected. It's just that my initial expectation was wrong 😅. Thank you!

Just one extra follow-up question on @davidepalazzo 's comment:

> A potential drawback of using GoogleSignin.signOut() is that it will also sign the user out from their default browser if they were logged in.

I do want the user to be prompted for selecting their Google account again (when signing in) after signing out. So I was thinking about calling `GoogleSignin.signOut()` from my mobile RN app to make sure that happens. Let's say they have more than 1 account on their device. If I don't allow them to select the account, from what I understood, they'll be stuck forever with signing in with the 1st account that they initially used, unless I call `GoogleSignin.signOut()`.

But does that really mean that user will be signed out from their default **browser**?

Thank you
